.github/workflows/test.yml

```
@@ -156,27 +156,7 @@ jobs:
-              echo ""
-              echo "=== Retrying $NUM_FAILED failed test(s): $FAILED ==="
-              echo ""
-              /bin/bash mfc.sh test -v --max-attempts 3 -j "$(nproc)" --only $FAILED $TEST_ALL || exit $?
-            else
-              echo "Too many failures ($NUM_FAILED) to retry — likely a real issue."
-              exit 1
-            fi
-          elif [ "$TEST_EXIT" -ne 0 ]; then
-            exit $TEST_EXIT
-          fi
+        run: bash .github/scripts/run-tests-with-retry.sh -v --max-attempts 3 -j "$(nproc)" $TEST_ALL $TEST_PCT
```

actionlint (1.7.11)

[error] shellcheck reported issue in this script: SC2086:info:1:80: Double quote to prevent globbing and word splitting

(shellcheck)